### PR TITLE
Add auth docs for sso enforced

### PR DESCRIPTION
**Pull request by technicallyjosh**
## Description

Adds missing text for auth endpoint.

## Reviews

- [ ] Technical review
- [x] Editorial review
- [ ] Product review

### CODE REVIEW

File: content/reference/api/hub/latest.yaml [hunk 1]
```diff
@@ -118,15 +118,19 @@ paths:
     post:
       tags:
         - authentication
-      summary: Create an authentication token
+      summary: Create an authenticated token
```
Comment (dvdksn) [inline content/reference/api/hub/latest.yaml:121]: I prefer the previous version; it is for authenticating to Docker Hub.

Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:121]: > I prefer the previous version; it is for authenticating to Docker Hub.

The old version is slightly wrong. We are trying to fix verbiage in many places. This route authenticated you. There is no more authentication to be done.

Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:121]: If it helps, this route is gonna be deprecated in the mid-future.

Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:121]: I'm going to just keep it as is... after thinking a bit, we a lot of work to do on this as an auth team and new APIs are coming in to replace.

**Review by thaJeztah — COMMENTED** [review]
Was there a conclusion on the new `403` status code returned? Was that in error, or intentional? If it's intentional, then a `403` should probably be added to this endpoint.

The overall response is a bit ambiguous; I see a `401` can _either_ be a "OK" response (2FA needed, so the resulting response must be handled, not just printed) or another error (in which case likely "just print the error"); https://github.com/docker/docs/blob/51cc027d864336442e2f04af1f7e021d8e877795/content/reference/api/hub/latest.yaml#L1138-L1156

Perhaps that logic should be documented in the endpoint as well 🤔

File: content/reference/api/hub/latest.yaml [hunk 2]
```diff
@@ -126,7 +126,11 @@ paths:
 
         The returned token is used in the HTTP Authorization header like `Authorization: Bearer {TOKEN}`.
 
-        Most Docker Hub APIs require this token either to consume or to get detailed information. For example, to list images in a private repository.
+        Most Docker Hub APIs require this token either to consume or to get detailed information. For example, to list 
+        images in a private repository.
+        
+        _**As of Monday, September 16, 2024, this route requires a PAT instead of a password if your organization has 
+        SSO enforced.**_
```
[the remark this answers is not in the document]
Comment (dvdksn) [reply content/reference/api/hub/latest.yaml:133]: There's a general Docker Hub "release notes" page where we can call this out: https://docs.docker.com/docker-hub/release-notes/

**Comment by technicallyjosh** [comment]
> Was there a conclusion on the new `403` status code returned? Was that in error, or intentional? If it's intentional, then a `403` should probably be added to this endpoint.
> 
> The overall response is a bit ambiguous; I see a `401` can _either_ be a "OK" response (2FA needed, so the resulting response must be handled, not just printed) or another error (in which case likely "just print the error");
> 
> https://github.com/docker/docs/blob/51cc027d864336442e2f04af1f7e021d8e877795/content/reference/api/hub/latest.yaml#L1138-L1156
> 
> Perhaps that logic should be documented in the endpoint as well 🤔

When using status codes for things, you really don't get the luxury of determining "what" caused it. If you look at it your way, they are all ambiguous and overloaded. 403 should not have been used on this legacy endpoint. These changes are already done on the API.

File: content/reference/api/hub/latest.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:133]: > does this apply to any endpoint using that type, or only to this endpoint?

Can you clarify this?

[the remark this answers is not in the document]
Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:133]: Keep in mind this is a disclaimer on this specific route due to changed behavior. We do not have a changelog currently due to these APIs not really being a product currently. We are working on fixing that when we deprecate all of these.

[the remark this answers is not in the document]
Comment (technicallyjosh) [reply content/reference/api/hub/latest.yaml:133]: Happy to have this convo outside of this simple change PR.

**Review by craig-osterhout — APPROVED** [review]

**Review by dvdksn — APPROVED** [review]

**Pull request merged**